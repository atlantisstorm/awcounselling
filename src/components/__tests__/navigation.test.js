import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Navigation from '../navigation';

describe('Navigation component', () => {
  it('should match snapshot', () => {
    const onClick = jest.fn();
    const selectedPage = "home";
    const { asFragment } = render(
      <Navigation onClick={ onClick } selectedPage={selectedPage} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain contact opton link', () => {
    const onClick = jest.fn();
    const selectedPage = "home";
    let link;
    const { getByTestId } = render(
      <Navigation onClick={ onClick } selectedPage={ selectedPage } />
    );

    link = getByTestId("navigation-home");
    const classAttrib = link.getAttribute("class");
    expect(classAttrib).toContain("active");

    const anchor = link.querySelector('a');
    fireEvent.click(anchor);
    expect(onClick).toHaveBeenCalledTimes(1);

    const dataName = anchor.getAttribute("data-name");
    expect(dataName).toBe("home");
  });
});