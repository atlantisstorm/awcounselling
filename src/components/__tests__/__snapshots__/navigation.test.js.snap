// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Aine Wilson Counselling
    </a>
    <button
      class="navbar-toggler"
      data-target="#navbarSupportContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarSupportContent"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item active"
          data-testid="navigation-home"
        >
          <a
            class="nav-link"
            data-name="home"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
          data-testid="navigation-about-me"
        >
          <a
            class="nav-link"
            data-name="about-me"
            href="/"
          >
            About Me
          </a>
        </li>
        <li
          class="nav-item"
          data-testid="navigation-family-programme"
        >
          <a
            class="nav-link"
            data-name="family-programme"
            href="/"
          >
            Family Programme
          </a>
        </li>
        <li
          class="nav-item"
          data-testid="navigation-timeout-programme"
        >
          <a
            class="nav-link"
            data-name="timeout-programme"
            href="/"
          >
            Time out Programme
          </a>
        </li>
        <li
          class="nav-item"
          data-testid="navigation-qualifications"
        >
          <a
            class="nav-link"
            data-name="qualifications"
            href="/"
          >
            Qualifications
          </a>
        </li>
        <li
          class="nav-item"
          data-testid="navigation-contact"
        >
          <a
            class="nav-link"
            data-name="contact"
            href="/"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
